pulsar/producer_partition.go

```diff
@@ -489,14 +488,13 @@ func (p *partitionProducer) internalSend(request *sendRequest) {
 
 	// The block chan must be closed when returned with exception
 	defer request.stopBlock()
-	if !p.canAddToQueue(request, uncompressedPayloadSize) {
+	if !p.canAddToQueue(request) {
```

Comment:
forget to runCallback

Reply:
Semaphore is released by `canReserveMem` if it failed.